Tests/SwiftCurrent_SwiftUITests/SwiftCurrent_SwiftUITests.swift

```diff
@@ -529,6 +530,32 @@ final class SwiftCurrent_SwiftUIConsumerTests: XCTestCase {
         wait(for: [expectOnAbandon1, expectOnAbandon2], timeout: TestConstant.timeout)
     }
 
+    func testWorkflowCanAbandonFromAPublisher() async throws {
+        let publisher = PassthroughSubject<Void, Never>()
```

Comment:
So other ways of using this would be to have an error publisher that's Output is Error and Failure is Never.  Then you could tie your workflow to automatically abandon if the publisher publishes an error.

I think that means that in general you subscribe to the publisher you really care about in your code, process the results and then send from the publisher you gave to the workflow.  Something like: RealPublisher -> ProcessPublisher -> Workflow publisher.

I'm trying to think through how this would effectively be used in the wild and how this could address the issue you are closing.

Reply:
So I'll state it more like this:
> The publisher you can pass to WorkflowView cannot have a failure type, because there's no logical behavior for dealing with errors.

What this means:
If you use a publisher (the "Real Publisher") that doesn't fail, you're already good. An example might be a publisher you have tied into button presses or observed objects with `@Published` properties. This covers an already large number of publishers you might be using. 

What's cool is that the `WorkflowView` will abandon when any value is emitted, so you don't have to do much other than hook it up to something that emits values.

What if your publisher can fail?
Then *you* must add explicit logic. One way to do that is simply to transform the error. For example I could take the "Real" publisher and call `.replaceError(with: <SomeValue>)`. 

Alternatively, maybe what you want is to trigger off of state. for that you could use `onChange`. In my example of abandoning a workflow when a user is logged out you could do something like.

```swift
VStack {
    Text("My View")
}
.onChange(User.isLoggedIn) { 
    if !$0 { 
        abandonPublisher.send() 
    } 
}
```

For this you need the setup you were just describing, a publisher you create yourself.

Does that help?

Reply:
Yup thank you!